easy-ui-react/src/VerticalNav/Subnav.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { Key, createContext, useContext, useMemo } from "react";
+import { ListProps, useListState } from "react-stately";
+import { SubnavItem } from "./SubnavItem";
+
+import styles from "./Subnav.module.scss";
+
+export type SubnavProps = {
+  /**
+   * The currently selected key in the navigation list.
+   */
+  selectedKey?: Key;
+
+  /**
+   * List of navigation items within the subnavigation.
+   */
+  children?: ListProps<object>["children"];
+};
+
+const SubnavLevelContext = createContext<number>(0);
+
+/**
+ * Represents a nested navigation within a navigation item.
+ */
+export function Subnav(props: SubnavProps) {
+  const { selectedKey, ...listStateProps } = props;
+  const levelContext = useContext(SubnavLevelContext);
+  const level = useMemo(() => levelContext + 1, [levelContext]);
```

Comment:
**question** what is level capturing here?

Reply:
level is capturing the nesting hierarchy for subnavigations that are within other subnavigations. the reason for this is that some styles are different depending on if it's a subnavigation or a subnavigation within a subnavigation.

let me know if that makes sense, or if there's an alternate name for this to better capture what it does

Reply:
ah makes sense